src/main/java/org/folio/helper/PurchaseOrderHelper.java

```diff
@@ -512,6 +509,65 @@ private CompletableFuture<Void> setPoNumberIfMissing(CompositePurchaseOrder comp
     return completedFuture(null);
   }
 
+  private CompletionStage<Void> setPoNumberPrefix(CompositePurchaseOrder compPO) {
+    if(Objects.nonNull(compPO.getPoNumberPrefix())) {
+      compPO.setPoNumber(compPO.getPoNumberPrefix() + compPO.getPoNumber());
+    }
+    return completedFuture(null);
+  }
+
+  private CompletionStage<Void> setPoNumberSuffix(CompositePurchaseOrder compPO) {
+    if(Objects.nonNull(compPO.getPoNumberSuffix())) {
+      compPO.setPoNumber(compPO.getPoNumber() + compPO.getPoNumberSuffix());
+    }
+    return completedFuture(null);
+  }
+
+  public CompletableFuture<Void> updatePrefixAndSuffix(CompositePurchaseOrder poFromStorage, CompositePurchaseOrder updatedPo) {
+    String poNumber = "";
+    if(isPrefixChanged(poFromStorage, updatedPo)) {
+      if(Objects.nonNull(updatedPo.getPoNumberPrefix()) && !updatedPo.getPoNumberPrefix().isEmpty()) {
+        if(Objects.nonNull(poFromStorage.getPoNumberPrefix()) && updatedPo.getPoNumber().contains(poFromStorage.getPoNumberPrefix())) {
+          poNumber = updatedPo.getPoNumber().replaceAll(poFromStorage.getPoNumberPrefix(), "");
+          updatedPo.setPoNumber(updatedPo.getPoNumberPrefix() + poNumber);
+        }
+        else {
+          updatedPo.setPoNumber(updatedPo.getPoNumberPrefix() + updatedPo.getPoNumber());
```

Comment:
nitpick: following the logic of the additions to this file, this line would be replaced by the `setPoNumberPrefix` defined above.

Reply:
fixed